### PR TITLE
cherry pick 76fab26c5c02cef38924d04136407489fd9457d9 from apache

### PR DESCRIPTION
**Pull request by mattp-altiscale**

### CODE REVIEW

File: hadoop-common-project/hadoop-common/CHANGES.txt [hunk 1]
```diff
@@ -2,11 +2,16 @@ Hadoop Change Log
 
 Release 2.7.2 - UNRELEASED
 
+Release 2.7.2 - 2016-01-25
```
Comment (altidaddy) [inline hadoop-common-project/hadoop-common/CHANGES.txt]: Nopes! You've cherry-picked into Altiscale's 2.7.1 release... So just add the line in OUR 2.7.1's Improvements section

**Comment by altidaddy** [comment]
Was it a clean cherry-pick other than the CHANGES.txt conflicts? If so, please feel free to self-merge

**Comment by altidaddy** [comment]
altidaddy == ravi's test github account btw

**Comment by mattp-altiscale** [comment]
tested via the following commands on local dev env:

local file:

s3cmd ls s3://com.altiscale.hadoop-test-s3/
                       DIR   s3://com.altiscale.hadoop-test-s3//
                       DIR   s3://com.altiscale.hadoop-test-s3/user/
2016-02-15 15:46         0   s3://com.altiscale.hadoop-test-s3/user_$folder$

hadoop distcp -Dhadoop.security.credential.provider.path=localjceks://file/Users/mattp/dev/HADOOP/aws.jceks -libjars /Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-core-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-annotations-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-databind-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-aws-2.7.1.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-core-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-kms-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-s3-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/joda-time-2.8.1.jar hdfs://localhost:9000/user/mattp/crib_sheet s3a://com.altiscale.hadoop-test-s3/

s3cmd ls s3://com.altiscale.hadoop-test-s3/
                       DIR   s3://com.altiscale.hadoop-test-s3//
                       DIR   s3://com.altiscale.hadoop-test-s3/user/
2016-02-29 18:15      6229   s3://com.altiscale.hadoop-test-s3/crib_sheet
2016-02-15 15:46         0   s3://com.altiscale.hadoop-test-s3/user_$folder$

s3cmd rm s3://com.altiscale.hadoop-test-s3/crib_sheet
File s3://com.altiscale.hadoop-test-s3/crib_sheet deleted

hdfs file:

s3cmd ls s3://com.altiscale.hadoop-test-s3/
                       DIR   s3://com.altiscale.hadoop-test-s3//
                       DIR   s3://com.altiscale.hadoop-test-s3/user/
2016-02-15 15:46         0   s3://com.altiscale.hadoop-test-s3/user_$folder$

hadoop distcp -Dhadoop.security.credential.provider.path=jceks://hdfs@localhost:9000/user/mattp/aws.jceks -libjars /Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-core-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-annotations-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/jackson-databind-2.2.3.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-aws-2.7.1.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-core-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-kms-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/aws-java-sdk-s3-1.10.6.jar,/Users/mattp/dev/hadoop/hadoop-dist/target/hadoop-2.7.1/share/hadoop/tools/lib/joda-time-2.8.1.jar hdfs://localhost:9000/user/mattp/crib_sheet s3a://com.altiscale.hadoop-test-s3/

s3cmd ls s3://com.altiscale.hadoop-test-s3/
                       DIR   s3://com.altiscale.hadoop-test-s3//
                       DIR   s3://com.altiscale.hadoop-test-s3/user/
2016-02-29 18:20      6229   s3://com.altiscale.hadoop-test-s3/crib_sheet
2016-02-15 15:46         0   s3://com.altiscale.hadoop-test-s3/user_$folder$

s3cmd rm s3://com.altiscale.hadoop-test-s3/crib_sheet

**Comment by raviprak-altiscale** [comment]
Sorry! I shouldn't have closed this PR. (I thought it was the one for dtutil)

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java [hunk 2]
```diff
@@ -109,10 +112,12 @@
...
-  public static final String SERVER_SIDE_ENCRYPTION_ALGORITHM = 
+  public static final String SERVER_SIDE_ENCRYPTION_ALGORITHM =
     "fs.s3a.server-side-encryption-algorithm";
-  
+
   public static final String S3N_FOLDER_SUFFIX = "_$folder$";
   public static final String FS_S3A_BLOCK_SIZE = "fs.s3a.block.size";
   public static final String FS_S3A = "s3a";
+
+  public static final int S3A_DEFAULT_PORT = -1;
```
Comment (raviprak-altiscale) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java:125]: This is a change I do not see in the open-source commit. Could you please explain why we need this?

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java [hunk 3]
```diff
@@ -979,6 +1066,9 @@ public S3AFileStatus getFileStatus(Path f) throws IOException {
 
         return new S3AFileStatus(true, false,
             f.makeQualified(uri, workingDir));
+      } else if (key.isEmpty()) {
+        LOG.debug("Found root directory");
```
Comment (raviprak-altiscale) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java:1074]: This is a change I do not see in the open-source commit. Could you please explain why we need this?

**Comment by raviprak-altiscale** [comment]
I cleaned up the messages and pushed the changes via https://github.com/Altiscale/hadoop/pull/30 . Thanks Matt

**Pull request closed without merging**